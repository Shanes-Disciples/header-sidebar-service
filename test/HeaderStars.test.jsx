import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import HeaderStars from '../client/src/components/HeaderStars.jsx';

configure({ adapter: new Adapter() });

describe('<HeaderStars />', () => {
  it('renders correctly', () => {
    const HeaderStarsComp = renderer.create(<HeaderStars avgRating={ 3.5 }/>).toJSON();
    expect(HeaderStarsComp).toMatchSnapshot();
  });
  it('renders 5 <FontAwesomeIcon /> stars', () => {
    const wrapper = shallow(<HeaderStars />);
    expect(wrapper.find(FontAwesomeIcon).length).toEqual(5);
  });
});
