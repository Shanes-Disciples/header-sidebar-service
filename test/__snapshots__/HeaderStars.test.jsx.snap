// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderStars /> renders correctly 1`] = `
<span>
  <span
    className="stars"
  >
    <span
      className="fullStar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="fullStar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="fullStar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="partialStar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <span
      className="emptyStar"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </span>
  <svg
    height="0"
    width="0"
  >
    <linearGradient
      id="starPartialFill"
    >
      <stop
        offset="0%"
        stopColor="#f4c150"
      />
      <stop
        offset="50%"
        stopColor="#f4c150"
      />
      <stop
        offset="50%"
        stopColor="#dedfe0"
      />
      <stop
        offset="100%"
        stopColor="#dedfe0"
      />
    </linearGradient>
    <linearGradient
      id="starEmpty"
    >
      <stop
        offset="100%"
        stopColor="#dedfe0"
      />
    </linearGradient>
    <linearGradient
      id="starFilled"
    >
      <stop
        offset="100%"
        stopColor="#f4c150"
      />
    </linearGradient>
  </svg>
</span>
`;
